approval_polls/views.py

```diff
@@ -1,101 +1,212 @@
+    template_name = 'approval_polls/detail.html'
+
+    def get_queryset(self):
+        return Poll.objects.filter(pub_date__lte=timezone.now())
+
 
 class ResultsView(generic.DetailView):
-  model = Poll
-  template_name = 'approval_polls/results.html'
-  def get_queryset(self):
-      return Poll.objects.filter(pub_date__lte=timezone.now())
+    model = Poll
+    template_name = 'approval_polls/results.html'
+
+    def get_queryset(self):
+        return Poll.objects.filter(pub_date__lte=timezone.now())
+
 
 @require_http_methods(['POST'])
 def vote(request, poll_id):
-  poll = get_object_or_404(Poll, pk=poll_id)
-  ballot = poll.ballot_set.create(timestamp=timezone.now())
+    poll = get_object_or_404(Poll, pk=poll_id)
+    ballot = poll.ballot_set.create(timestamp=timezone.now())
+
+    for counter, choice in enumerate(poll.choice_set.all()):
+        if 'choice' + str(counter + 1) in request.POST:
+            ballot.vote_set.create(choice=choice)
+            ballot.save()
 
-  for counter, choice in enumerate(poll.choice_set.all()):
-    if 'choice' + str(counter + 1) in request.POST:
-      ballot.vote_set.create(choice = choice)
-      ballot.save()
+    poll.save()
 
-  poll.save()
+    return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
-  return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
 def embed_instructions(request, poll_id):
-  link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
-  return render(request, 'approval_polls/embed_instructions.html', {'link': link})
+    link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
+    return render(request, 'approval_polls/embed_instructions.html', {'link': link})
 
-class CreateView(generic.View):
 
-  @method_decorator(login_required)
-  def get(self, request, *args, **kwargs):
-    return render(request, 'approval_polls/create.html')
+class PollUpdate(generic.View):
+    '''A generic update view for our poll.
+
+    Since this is actually a compound form updating two different models
+    a Poll and "n" number of Choice instances we cannot use the straight forward
+    UpdateView.
+    '''
+
+
+    success_url = '/approval_polls/my-polls'
+
+    @method_decorator(login_required)
+    def get(self, request, pk):
+        '''We have to get the object and the choices.
+
+        Arguments:
+            request (WSGIRequest): the standard request that django calls the get method with
+            pk (int): the id of the poll object we are mean to edit
+
+        Returns:
+            HTTPResponse
+        '''
+
+        form = PollUpdateForm(instance=Poll.objects.get(pk=pk))
+
+        choices = [{'form': ChoiceForm(instance=choice), 'id': choice.id} for choice in Choice.objects.filter(poll__pk=pk)]
+        my_context = {'form': form, 'choices': choices}
+        request_context = RequestContext(request)
+
+        return render_to_response('approval_polls/poll_update_form.html',
```

Comment:
OH wow yea. I've been using Django for quite a while. Since before render existed. :)

render is just a shortcut that automagically puts in the request context for you. Other than that the functions are identical. 

I'm just in the habit of using render_to_repsonse. I can change that for consistency.